packages/js/tsup.config.ts

```diff
@@ -0,0 +1,24 @@
+import { defineConfig, Options } from 'tsup';
+
+const baseConfig: Options = {
+  splitting: false,
+  sourcemap: false,
+  clean: true,
+  dts: true,
+};
+
+export default defineConfig([
+  { ...baseConfig, entry: ['src/index.ts'], format: ['esm', 'cjs'] },
+  {
+    ...baseConfig,
+    entry: { novu: 'src/umd.ts' },
+    format: ['iife'],
+    minify: true,
+    dts: false,
+    outExtension: () => {
+      return {
+        js: '.min.js',
+      };
+    },
```

Comment:
Let's also add the following:

// @ts-ignore
import { name, version } from './package.json';

1. Trick to inject version and name in the source code without importing the full package.json
```
  define: {
      PACKAGE_NAME: `"${name}"`,
      PACKAGE_VERSION: `"${version}"`,
      __DEV__: `${!isProd}`,
    },
```

2. To fully support `ESM/CJS` we need to copy a package.json with the necessary `ESM/CJS` attributes into each dist folder [See example](https://github.com/clerk/javascript/blob/main/packages/nextjs/tsup.config.ts#L40).

Reply:
1. Ok
2. Not sure I understand why we need this, and we have all esm/cjs files in one dist folder